### PR TITLE
no ttl for aggregates

### PR DESCRIPTION
**Pull request by gsps1**
for aggregate resolution , we skip adding TTL
https://issues.cask.co/browse/CDAP-1626

### CODE REVIEW

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricDatasetFactory.java [hunk 1]
```diff
@@ -80,7 +80,8 @@ public FactTable get(int resolution) {
     int ttl =  cConf.getInt(MetricsConstants.ConfigKeys.RETENTION_SECONDS + "." + resolution + ".seconds", -1);
 
     DatasetProperties.Builder props = DatasetProperties.builder();
-    if (ttl > 0) {
+    // don't add TTL for MAX_RESOLUTION table.
+    if (ttl > 0 && resolution != Integer.MAX_VALUE) {
```
Comment (abaranau) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricDatasetFactory.java]: Is there an easy way to make that configurable? I see above lines try to get it from config. Can we default it in config to unlim or is it toodangerous?

Comment (gsps1) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricDatasetFactory.java]: by default if TTL is not set , the TTL would be forever (INT_MAX) , if someone sets a lower TTL for the totals table, we want to ignore it and use TTL forever ( according to CDAP-1626). if we want to allow that to be configurable, then the issue and PR is not relevant right ?

Comment (abaranau) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricDatasetFactory.java]: alright - let's override it for now. Can you add in that comment mention about jira issue "CDAP-1626" please.

**Comment by abaranau** [comment]
adjust a comment and merge please!

**Pull request merged**